code/tasks/LDAPGroupSyncTask.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Class LDAPGroupSyncTask
+ *
+ * A task to sync all groups to the site using LDAP.
+ */
+class LDAPGroupSyncTask extends BuildTask {
+
+	private static $dependencies = array(
+		'ldapService' => '%$LDAPService'
+	);
+
+	/**
+	 * Setting this to true causes the sync to delete any local Group
+	 * records that were previously imported, but no longer existing in LDAP.
+	 *
+	 * @config
+	 * @var bool
+	 */
+	private static $destructive = false;
+
+	public function getTitle() {
+		return _t('LDAPGroupSyncJob.SYNCTITLE', 'Sync all groups from Active Directory');
+	}
+
+	public function run($request) {
+		// get all groups from LDAP, but only get the attributes we need.
+		// this is useful to avoid holding onto too much data in memory
+		// especially in the case where getGroups() would return a lot of groups
+		$ldapGroups = $this->ldapService->getGroups(
+			false,
+			array('objectguid', 'samaccountname', 'dn'),
+			// Change the indexing attribute so we can look up by GUID during the deletion process below.
+			'objectguid'
+		);
+
+		$start = time();
+
+		$count = 0;
+
+		foreach($ldapGroups as $data) {
+			$group = Group::get()->filter('GUID', $data['objectguid'])->limit(1)->first();
+
+			if(!($group && $group->exists())) {
+				// create the initial Group with some internal fields
+				$group = new Group();
+				$group->GUID = $data['objectguid'];
+
+				$this->log(sprintf(
+					'Creating new Group (ID: %s, GUID: %s, sAMAccountName: %s)',
+					$group->ID,
+					$data['objectguid'],
+					$data['samaccountname']
+				));
+			} else {
+				$this->log(sprintf(
+					'Updating existing Group "%s" (ID: %s, GUID: %s, sAMAccountName: %s)',
+					$group->getTitle(),
+					$group->ID,
+					$data['objectguid'],
+					$data['samaccountname']
+				));
+			}
+
+			$group->Title = $data['samaccountname'];
```

Comment:
So the `name` is now used as `Title` (this is the group name in AD). `samaccountname` was shifted into `Code` (this is Windows-2000 account name, akin to Member's Username). Also added `description` sync, since we have that fields on both sides too.